packages/node-core/src/indexer/connectionPool.service.ts

```diff
@@ -71,6 +170,84 @@ export class ConnectionPoolService<T extends ApiConnection> implements OnApplica
     }
 
     logger.info(`reconnected to ${endpoint}!`);
-    this.connectionPool[apiIndex] = this.allApi[apiIndex];
+  }
+
+  private calculatePerformanceScore(responseTime: number, failureCount: number): number {
+    const responseTimeScore = 1 / responseTime;
+    const failureScore = 1 - failureCount / MAX_FAILURES;
+    return RESPONSE_TIME_WEIGHT * responseTimeScore + FAILURE_WEIGHT * failureScore;
+  }
+
+  @Interval(LOG_INTERVAL_MS)
+  logEndpointStatus(): void {
```

Comment:
It would be good if we could log the weights of the endpoints for debugging purposes